### PR TITLE
feat: workspace create front and backend

### PR DESCRIPTION
**Pull request by RedCrazyGhost**
### Description

add workspace front code
add workspace create dao transaction code
add workspace get list data

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible

### CODE REVIEW

File: apps/workbench/backend/router/router.go [hunk 1]
```diff
@@ -28,7 +28,12 @@ func mountAPIs(e *gin.Engine, ctx context.ApplicationContext) {
 		api.POST("/forgot_verify", developer.ForgotVerifySend(ctx))
 		api.PUT("/reset_password", developer.ResetPassword(ctx))
 		//api.PUT("/developer/username/:id")
-		api.POST("/workspace/create", middleware.JWTAuthMiddleware(), workspace.WorkspaceCreate(ctx))
+	}
+	workspaceApi := api.Group("/workspace")
+	{
+		workspaceApi.POST("/create", middleware.JWTAuthMiddleware(), workspace.WorkspaceCreate(ctx))
+		workspaceApi.POST("/list", middleware.JWTAuthMiddleware(), workspace.WorkspaceGetListById(ctx))
+		workspaceApi.POST("/logo", workspace.WorkspaceGetLogoPath(ctx))
```
Comment (zccabb) [inline apps/workbench/backend/router/router.go:36]: fix router ,not exist  api verison, elegant and practical middleware

File: apps/workbench/backend/api/workspace/workspace.go [hunk 2]
```diff
@@ -30,6 +30,60 @@ type WorkspaceCreateResp struct {
...
+				Msg:  "upload workspace logo file unusual.",
+				Data: nil,
+			})
+			return
+		}
+
+		c.AbortWithStatusJSON(http.StatusOK, types.NewValidResponse(&Resp{
+			Code: e.Success,
+			Msg:  "upload workspace logo succeed",
+			Data: path,
```
Comment (zccabb) [inline apps/workbench/backend/api/workspace/workspace.go:60]: Should the address of the image server be exposed?

**Review by zccabb — COMMENTED** [review]
fix your pr title

**Comment by yuluo-yx** [comment]
pls fix ci and solve conflicts. @RedCrazyGhost

File: apps/workbench/backend/router/router.go (hunk 1 above)
Comment (RedCrazyGhost) [reply apps/workbench/backend/router/router.go:36]: The actual router path is /v1/workspace/create

File: apps/workbench/backend/api/workspace/workspace.go (hunk 2 above)
Comment (RedCrazyGhost) [reply apps/workbench/backend/api/workspace/workspace.go:60]: add jwt authentication

**Review by zccabb — APPROVED** [review]

**Review by yuluo-yx — APPROVED** [review]

**Pull request merged**